windows/src/desktop/kmshell/install/UfrmInstallKeyboard.pas

```diff
@@ -513,8 +513,10 @@ function TfrmInstallKeyboard.InstallTipForKeyboard(const BCP47Tag: string): Bool
 begin
   // Ensure keyboard is recorded in CU registry before we install the TIP, otherwise it will be marked as disabled by default,
   // as installing the TIP adds CU registry settings, potentially confusing the keyboard settings
-  kmcom.Refresh;
-  kmcom.Apply;
+
+  // Only do this for the keyboards, so we don't cause a global refresh
+  kmcom.Keyboards.Refresh;
+  kmcom.Keyboards.Apply;
```

Comment:
I see a lot of places with `kmcom.Refresh` and `kmcom.Apply` in the new code... and a lot of places with `kmcom.Keyboards.Refresh` and `kmcom.Keyboards.Apply`.  Some of the places use half of each pair.

What determines which variant is used in a given circumstance?

Reply:
This is the result of an unfortunate API side-effect. `kmcom.Apply` not only writes the changes out, it also notifies all apps on the system of the changes. However, `kmcom.Keyboards.Apply` only writes the changes for the keyboards and does not do the global notification. So when we are midway through a process, often we'll just `Apply` the area we know has changed, to reduce chatter and unnecessary multiple notifications.